compiler/AST/AstToText.cpp

```diff
@@ -1309,6 +1442,76 @@ void AstToText::appendExpr(NamedExpr* expr, bool printingType)
   appendExpr(expr->actual, printingType);
 }
 
+void AstToText::appendExpr(IfExpr* expr, bool printingType)
+{
+  mText += "if ";
+  if (Expr* sel = toExpr(expr->getCondition()))
+  {
+    appendExpr(sel, printingType);
+  }
+  mText += " then ";
+  if (BlockStmt* thenBlockStmt = toBlockStmt(expr->getThenStmt()))
+  {
+    Expr* exp = thenBlockStmt->body.get(1);
```

Comment:
It'd be good to assert we aren't dropping code on the floor here and in the else branch